market_nextdoor_api/views/preorder_views.py

```diff
@@ -62,4 +62,38 @@ def update_preorder(preorder, data):
 
 def delete_preorder(preorder):
   preorder.delete()
-  return Response(status=status.HTTP_204_NO_CONTENT)
\ No newline at end of file
+  return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+#manytomany views testing
+@api_view(['GET', 'POST'])
+def preorder_test_list(request, customer_id):
+  try:
+    check_customer = Customer.objects.get(pk=customer_id)
+  except Customer.DoesNotExist:
+    return Response(status=status.HTTP_404_NOT_FOUND)
+
+  if request.method == 'GET':
+    return get_preorder_test_list(request, check_customer)
+  elif request.method == 'POST':
+    return create_preorder_test(request, check_customer)
+
+def get_preorder_test_list(request, check_customer):
+  preorder_tests = Preorder_test.objects.filter(customer=check_customer)
+  serializer = Preorder_testSerializer(preorder_tests, many=True)
+  return Response(serializer.data)
+
+def create_preorder_test(request, check_customer):
+  serializer = Preorder_testSerializer(data=request.data)
+  if serializer.is_valid():
+    preorder = serializer.save()
+
+    items = request.data.get('items', [])
+    for item in items:
+      Preorder_testItem.objects.create(
+        preorder=preorder,
+        item_id=item['item'],
+        quantity=item['quantity']
+      )
+    return Response(serializer.data, status=status.HTTP_201_CREATED)
+  return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
would an 'else' statement here make sense? It seems as though this method will return a bad request as an alternative to everything else, which isn't technically incorrect, but could make it a bit more readable. Once again, newbie here making suggestions so probably wrong.

Reply:
Agreed, when we build out functionality for this on main, will definitely need to add further logic to avoid bad requests.